scripts/openwebnet-discovery.nse

```diff
@@ -134,6 +134,15 @@ local function get_response(sd, request)
 
   repeat
     status, data = sd:receive_buf("##", true)
+
+    if status == nil then
+      stdnse.debug("Error: " .. tostring(data))
+      -- Captures the NACK after TIMEOUT occured.
+      -- Avoids false results.
+      status, data = sd:receive_buf("##", true)
```

Comment:
This should only be attempted if the error was "TIMEOUT". The error could be "EOF" or something else, which means we can't read any further.